### ISSUE TITLE
Dict unpacking accepts invalid types

### ISSUE THREAD

**Issue by refi64** [astral-sh/ty#1497]
### Summary

https://play.ty.dev/6fed81b8-6589-4e8b-9adf-71bc1676d9cc

```python
x = {**1}
```

This succeeds without any errors, even though it of course fails at runtime. This seems to work just fine in list literals (show in the playground link) but not in dicts. Might be related to https://github.com/astral-sh/ty/issues/1332? But that seems to be more specifically about "mappings with the wrong types" rather than "not a mapping at all".

### Version

35640dd85

**Comment by AlexWaygood** [comment]
Thanks! Please see #1332 :-)

**Comment by AlexWaygood** [comment]
> Might be related to https://github.com/astral-sh/ty/issues/1332? But that seems to be more specifically about "mappings with the wrong types" rather than "not a mapping at all".

Specifically, this line in the example in #1332 covers your issue here:

```py
    reveal_type({**d})  # dict[Unknown, Unknown]             (good, but no diagnostic emitted!)
```

Since `d` is typed as `object` in that snippet (it is not a mapping at all)
